homeassistant/components/shelly/entity.py

```diff
@@ -586,19 +550,6 @@ def available(self) -> bool:
             self.coordinator.device.status[self.key][self.entity_description.sub_key]
         )
 
-    @property
-    def extra_state_attributes(self) -> dict[str, Any] | None:
-        """Return the state attributes."""
-        if self.entity_description.extra_state_attributes is None:
-            return None
-
-        assert self.coordinator.device.shelly
-
-        return self.entity_description.extra_state_attributes(
-            self.coordinator.device.status[self.key][self.entity_description.sub_key],
-            self.coordinator.device.shelly,
-        )
-
```

Comment:
Currently we don't have any RPC (Gen2) sensors which uses `extra_state_attributes` and since we try to minimize the use of them in core it is better to remove this code. This was used by old binary sensors update entities which migrated to use the update platform.